packages/melos/lib/src/common/environment_variable_key.dart

```diff
@@ -0,0 +1,44 @@
+class EnvironmentVariableKey {
```

Comment:
Wouldn't this be better as an enhanced enum?

Reply:
@spydon, initially, I did think of using an Enum for it, i.e.,
```dart
enum EnvironmentVariableKey {
  melosRootPath('MELOS_ROOT_PATH'),
  melosPackageName('MELOS_PACKAGE_NAME'),
  melosPackageVersion('MELOS_PACKAGE_VERSION');
  // the rest of the values...

  const EnvironmentVariableKey(this._value);

  final String _value;

  @override
  String toString() => _value;
}
```
but decided not to do so because of these 2 reasons:
1. A `String` is expected in every place the env values are used at the moment and with such an enum it is possible to access not only the value, i.e. `EnvironmentVariableKey.melosRootPath.toString()` or `EnvironmentVariableKey.melosRootPath.value`, if `value` made public, but also accidentally use a name of the value, i.e. `EnvironmentVariableKey.melosRootPath.name`. It seemed to me like a more error-prone approach for the current case.
2. Since strings are expected in all the cases, I'd have to either use a `toString` on the enum values or make the `value` public which results in longer lines compared to the approach with static const strings.
`EnvironmentVariableKey.melosRootPath` vs `EnvironmentVariableKey.melosRootPath.value` / `EnvironmentVariableKey.melosRootPath.toString()`

Does it sound reasonable?

Reply:
I think strings should be used as little as possible for these cases, isn't it possible to just have a `fromString` method in the enum and use it everywhere there are comparisons instead? Then there isn't any need for string comparisons, and it looks like you've already removed most of the string comparisons anyways?

Reply:
>  it looks like you've already removed most of the string comparisons anyways?

Not really, the env variable values are used as strings everywhere at the moment: in maps, i.e. `Map<String, String>`, in string interpolations, i.e. '%${EnvironmentVariableKey.melosScript}%'`, so @spydon, I do not see how we can easily use the Enum everywhere, at least without a way bigger refactoring.

@Salakar, would you like to share your opinion as well?

Reply:
Ah alright, let's merge this first then and then turn it into an enum then, this is one step in the right direction at least!

Reply:
Yep, agree, this is indeed a step to a more maintainable code base, let's get it merged then!